frodo.js

```diff
@@ -5,11 +5,11 @@
  * for the extras/ plugins if they are loaded into a presentation.
```

Comment:
SCRIPT-The way you've refactored this complex method into simpler ones is great. It makes the code much more maintainable.